src/gt4py/frontend/gtscript_frontend.py

```diff
@@ -654,56 +658,76 @@ def visit_If(self, node: ast.If):
-#
-#     def visit_FieldRef(self, node: nodes.FieldRef):
-#         assert node.name in self.defines
-#         self.reads[node.name] = self.reads.get(node.name, 0) + 1
-#
-#     def visit_FieldDecl(self, node: nodes.FieldDecl):
-#         assert node.is_api is False
-#         self.defines[node.name] = "temp"
-#
-#     def visit_Assign(self, node: nodes.Assign):
-#         if isinstance(node.target, nodes.FieldRef):
-#             for alias in self.aliases.get(node.target.name, set()):
-#                 self.aliases[alias] -= {node.target.name}
-#                 self.aliases[node.target.name] = set()
-#
-#             if isinstance(node.value, nodes.FieldRef):
-#                 if node.target.name not in itertools.chain(self.writes.keys(), self.reads.keys()):
-#                     aliases = set(self.aliases.setdefault(node.value.name, set()))
-#                     self.aliases.setdefault(node.target.name, set())
-#                     self.aliases[node.target.name] |= aliases | {node.value.name}
-#                     for alias in self.aliases[node.target.name]:
-#                         self.aliases[alias] |= {node.target.name}
-#
-#         self.visit(node.value)
-#         self.writes[node.target.name] = self.reads.get(node.target.name, 0) + 1
-#
-#     def visit_StencilDefinition(self, node: nodes.StencilDefinition):
-#         for decl in node.api_fields:
-#             self.defines[decl.name] = "api"
-#         for computation in node.computations:
-#             self.visit(computation)
-#
-#         for a in ["defines", "reads", "writes"]:
-#             print(f"{a}: ", getattr(self, a))
-#
+def _make_temp_decls(
+    descriptors: Dict[str, gtscript._FieldDescriptor]
+) -> Dict[str, nodes.FieldDecl]:
+    return {
+        name: nodes.FieldDecl(
+            name=name,
+            data_type=nodes.DataType.from_dtype(desc.dtype),
+            axes=[str(ax) for ax in desc.axes],
+            is_api=False,
+            data_dims=list(desc.data_dims),
+        )
+        for name, desc in descriptors.items()
+    }
+
+
+def _make_init_computations(
+    temp_decls: Dict[str, nodes.FieldDecl], init_values: Dict[str, Any], func_node: ast.AST
+) -> List[nodes.ComputationBlock]:
+    if not temp_decls:
+        return []
+
+    axes = set().union(*(set(temp_decls[name].axes) for name in temp_decls.keys()))
+    if "I" not in axes or "J" not in axes:
+        raise GTScriptSyntaxError(
+            message="Typed temporaries must to be defined on the I and J axis.",
+            loc=nodes.Location.from_ast_node(func_node),
+        )
+
+    if "K" in axes:
+        order = nodes.IterationOrder.PARALLEL
+        interval = nodes.AxisInterval.full_interval()
+    else:
+        order = nodes.IterationOrder.FORWARD
+        interval = nodes.AxisInterval(
+            start=nodes.AxisBound(level=nodes.LevelMarker.START, offset=0),
+            end=nodes.AxisBound(level=nodes.LevelMarker.START, offset=1),
+        )
```

Comment:
```suggestion
    order = nodes.IterationOrder.PARALLEL
    interval = nodes.AxisInterval.full_interval()
```
If we don't support 2D temporaries, this is not necessary and the above check can ensure all 3 dimensions are present right?